stdlib/REPL/src/TerminalMenus/Pager.jl

```diff
@@ -0,0 +1,45 @@
+mutable struct Pager{C} <: _ConfiguredMenu{C}
+    lines::Array{String,1}
+    pagesize::Int
+    pageoffset::Int
+    selected::Nothing
+    config::C
+end
+
+function Pager(text::AbstractString; pagesize::Int=10, kwargs...)
+    lines = readlines(IOBuffer(text))
+    return Pager(lines, pagesize, 0, nothing, Config(; kwargs...))
+end
+
+function header(p::Pager)
+    total = length(p.lines)
+    current = min(p.pageoffset + p.pagesize, total)
+    percent = round(Int, (current / total) * 100)
+    return "($(lpad(current, ndigits(total))) / $total) $(lpad(percent, 3))%"
+end
+
+options(p::Pager) = p.lines
+
+cancel(::Pager) = nothing
+
+pick(::Pager, ::Int) = true
+
+function writeline(buf::IOBuffer, pager::Pager{Config}, idx::Int, iscursor::Bool)
+    print(buf, pager.lines[idx])
+end
+
+function writeLine(buf::IOBuffer, pager::Pager{<:Dict}, idx::Int, cursor::Bool)
```

Comment:
Since this is a new menu type, do you need to support the legacy interface? There wouldn't be any callers of it out in the wild, and no one using Julia 1.5 or earlier will be able to use the pager. So I think you can cleanly target the new API.

In case you're unfamiliar, see #35915.